backend/src/middlewares/auth/auth.middleware.ts

```diff
@@ -32,4 +35,26 @@ const localAuth = (
   )(req, res, next);
 };
 
-export { localAuth };
+const signUpMiddleware = async (
+  req: TypedRequestBody<Prisma.UserCreateInput>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    await signUpSchema.validate(req.body);
+    const existedUser = await prisma.user.findFirst({
+      where: {
+        email: req.body.email,
+      },
+    });
+    if (existedUser) throw new Error();
+
+    next();
+  } catch (err) {
+    console.error(err);
+    res.status(403);
+    next(new Error('Some error'));
```

Comment:
Some error have no logic context. You should handle errors

Reply:
I did this at first, because when Oleksandr said that it is better to always return a 403 error and so that the hacker does not understand the reason of failure, from a security point of view.
I fixed it, now it returns an error message.